FhirToDataLake/Synapse-search.yml

```diff
@@ -0,0 +1,210 @@
+      # Deploy pipeline at (0, 5, 10,...,55) so that after pipeline been created, it will not be triggered immediately
+      $secondsDelay = 5*60 - ([int](Get-Date -Format ss) + 60*([int](Get-Date -Format mm)%5))
+      Start-Sleep -s $secondsDelay
+
+      New-AzResourceGroupDeployment -Name $(deploymentName) -ResourceGroupName $(resourceGroup) -TemplateFile $(System.DefaultWorkingDirectory)/FhirToDataLake/deploy/templates/FhirSynapsePipelineTemplate.json -TemplateParameterObject $templateParameters -Verbose
+      $storageAccountName = (Get-AzResource -ResourceGroupName $(resourceGroup) -Name $(appName)* -ResourceType Microsoft.Storage/storageAccounts).Name
+      
+      # Get storage name and inject it into pipeline variable
+      Write-Host "##vso[task.setvariable variable=storageAccountName;]$storageAccountName"
+
+- task: AzurePowerShell@5
+  displayName: Fetch data from FHIR server
+  inputs: 
+    azureSubscription: 'ResoluteOpenSource'
+    azurePowerShellVersion: latestVersion
+    ScriptType: inlineScript
+    Inline: |
+      # Here delay some time so that function be triggered twice
+      Write-Host 'Delay 7 minutes to fetch data from FHIR server'
+      Start-Sleep -s 420
+
+      Write-Host 'Stop function at $(Get-Date -Format hh:MM:ss)'
+      Stop-AzFunctionApp -Name $(appName) -ResourceGroupName $(resourceGroup) -Force
+  
+- task: AzureCLI@2
+  displayName: Assign access premission roles
+  inputs: 
+    azureSubscription: 'ResoluteOpenSource'
+    scriptType: 'ps'
+    scriptLocation: 'inlineScript'
+    inlineScript: |
+      $subscriptionId = az account show --query 'id' -o tsv
+      $synapsePrincipalId = az synapse workspace show --name $(synapseWorkspaceName) --resource-group $(resourceGroup) --query 'identity.principalId' -o tsv
+      $serviceConnectionPrincipalId = az account show --query 'user.name' -o tsv
+
+      $storageScope = "/subscriptions/$($subscriptionId)/resourceGroups/$(resourceGroup)/providers/Microsoft.Storage/storageAccounts/$(storageAccountName)"
+
+      Write-Host 'Assign Storage Blob Data Contributor for Synapse Workspace to Storage.'
+      az role assignment create --role "Storage Blob Data Contributor" --assignee-object-id $synapsePrincipalId --scope $storageScope --assignee-principal-type ServicePrincipal
+      
+      Write-Host 'Assign Storage Blob Data Contributor for service connection account to Storage.'
+      az role assignment create --role "Storage Blob Data Contributor" --assignee $serviceConnectionPrincipalId --scope $storageScope
+
+      Write-Host 'Assign Synapse SQL Administrator for service connection account to Synapse SQL pool.'
+      az synapse role assignment create --workspace-name $(synapseWorkspaceName) --role "Synapse SQL Administrator" --assignee $serviceConnectionPrincipalId
+      
+      # Wait 1 min in case the assignments don't enable.
+      Start-Sleep -s 60
+
+- task: AzurePowerShell@5
+  displayName: Set up EXTERNAL TABLEs and VIEWs on Synapse.
```

Comment:
`Set up Synapse Environment` to keep consistent with the script name?

Reply:
Updated, thanks